### PR TITLE
Enable qemu debugging of guest VMs

### PR DESCRIPTION
**Pull request by ganeshmaharaj**
Signed-off-by: Ganesh Maharaj Mahalingam <ganesh.mahalingam@intel.com>

### CODE REVIEW

File: ccloudvm.go [hunk 1]
```diff
@@ -140,12 +140,13 @@ func (d *drives) Set(value string) error {
 	return nil
 }
 
-func vmFlags(fs *flag.FlagSet, memGB, CPUs *int, m *mounts, p *ports, d *drives) {
+func vmFlags(fs *flag.FlagSet, memGB, CPUs *int, m *mounts, p *ports, d *drives, Qemuport *uint) {
```
Comment (markdryan) [inline ccloudvm.go]: This one shouldn't be exported.  Just one in the VMSpec struct.

File: ccloudvm.go [hunk 2]
```diff
@@ -175,6 +176,7 @@ func createFlags(ctx context.Context, ws *workspace) (*workload, bool, error) {
 	var p ports
 	var d drives
 	var memGiB, CPUs int
+	var Qemuport uint
```
Comment (markdryan) [inline ccloudvm.go]: Same comment.  This shouldn't be capitalized.

File: README.md [hunk 3]
```diff
@@ -411,6 +411,18 @@ the workloads define the following in their cloudinit documents.
 ```
 package_upgrade: {{with .PackageUpgrade}}{{.}}{{else}}false{{end}}
 ```
+The -qemuport option can be used to follow qemu boot logs by connecting to user defined port or an entry in the workload yaml
+file.
+
+For example,
+
+qemuport: 9999
```
Comment (markdryan) [inline README.md]: This should be --qemuport=9999

File: ccloudvm.go [hunk 4]
```diff
@@ -140,12 +140,13 @@ func (d *drives) Set(value string) error {
...
 }
 
-func vmFlags(fs *flag.FlagSet, memGB, CPUs *int, m *mounts, p *ports, d *drives) {
+func vmFlags(fs *flag.FlagSet, memGB, CPUs *int, m *mounts, p *ports, d *drives, qemuport *uint) {
 	fs.IntVar(memGB, "mem", *memGB, "Gigabytes of RAM allocated to VM")
 	fs.IntVar(CPUs, "cpus", *CPUs, "VCPUs assigned to VM")
 	fs.Var(m, "mount", "directory to mount in guest VM via 9p. Format is tag,security_model,path")
 	fs.Var(d, "drive", "Host accessible resource to appear as block device in guest VM.  Format is path,format[,option]*")
 	fs.Var(p, "port", "port mapping. Format is host_port-guest_port, e.g., -port 10022-22")
+	fs.UintVar(qemuport, "qemuport", *qemuport, "Port to follow qemu logs of the guest machine, eg., -q 9999")
```
Comment (markdryan) [inline ccloudvm.go]: eg., -q 9999

should be

e.g., -qemuport 9999

File: vm.go [hunk 5]
```diff
@@ -168,6 +174,7 @@ func statusVM(ctx context.Context, instanceDir, keyPath, workloadName string, ss
 	fmt.Fprintf(w, "Workload\t:\t%s\n", workloadName)
 	fmt.Fprintf(w, "Status\t:\t%s\n", status)
 	fmt.Fprintf(w, "SSH\t:\t%s\n", ssh)
+	fmt.Fprintf(w, "QEMU Debug Port\t:\t%d\n", qemuport)
```
Comment (markdryan) [inline vm.go]: It would be better if we didn't print out this information if qemuport was 0.

File: ccloudvm.go (hunk 4 above)
Comment (ganeshmaharaj) [reply ccloudvm.go]: *facepalm*. WIll fix it.

**Comment by ganeshmaharaj** [comment]
Thanks. 😄

**Pull request merged**